subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DependencyManagementBuildScopeServices.java

```diff
@@ -698,9 +741,135 @@ public void buildFinished(GradleInternal build) {
-                                                              FeaturePreviews featurePreviews) {
-        return new DefaultDependenciesAccessors(registry, workspace, factory, buildOperationExecutor, featurePreviews);
+                                                              ExecutionEngine executionEngine,
+                                                              FeaturePreviews featurePreviews,
+                                                              FileCollectionFactory fileCollectionFactory,
+                                                              ClasspathFingerprinter fingerprinter) {
+        return new DefaultDependenciesAccessors(registry, workspace, factory, featurePreviews, executionEngine, fileCollectionFactory, fingerprinter);
+    }
+
+
+    /**
+     * Execution engine for usage above Gradle scope
+     *
+     * Currently used for running artifact transformations in buildscript blocks.
+     */
+    ExecutionEngine createExecutionEngine(
+            BuildOperationExecutor buildOperationExecutor,
+            ClassLoaderHierarchyHasher classLoaderHierarchyHasher,
+            Deleter deleter,
+            ExecutionStateChangeDetector changeDetector,
+            ListenerManager listenerManager,
+            OutputSnapshotter outputSnapshotter,
+            OverlappingOutputDetector overlappingOutputDetector,
+            TimeoutHandler timeoutHandler,
+            ValidateStep.ValidationWarningReporter validationWarningReporter,
+            ValueSnapshotter valueSnapshotter
+    ) {
+        OutputChangeListener outputChangeListener = listenerManager.getBroadcaster(OutputChangeListener.class);
+        // TODO: Figure out how to get rid of origin scope id in snapshot outputs step
+        UniqueId fixedUniqueId = UniqueId.from("dhwwyv4tqrd43cbxmdsf24wquu");
+        // @formatter:off
+        return new DefaultExecutionEngine(
+            new IdentifyStep<>(valueSnapshotter,
+            new IdentityCacheStep<>(
+            new AssignWorkspaceStep<>(
+            new LoadExecutionStateStep<>(
+            new ValidateStep<>(validationWarningReporter,
+            new CaptureStateBeforeExecutionStep(buildOperationExecutor, classLoaderHierarchyHasher, outputSnapshotter, overlappingOutputDetector, valueSnapshotter,
+            new NoOpCachingStateStep(
+            new ResolveChangesStep<>(changeDetector,
+            new SkipUpToDateStep<>(
+            new BroadcastChangingOutputsStep<>(outputChangeListener,
+            new StoreExecutionStateStep<>(
+            new SnapshotOutputsStep<>(buildOperationExecutor, fixedUniqueId, outputSnapshotter,
+            new CreateOutputsStep<>(
+            new TimeoutStep<>(timeoutHandler,
+            new ResolveInputChangesStep<>(
+            new RemovePreviousOutputsStep<>(deleter, outputChangeListener,
+            new ExecuteStep<>(buildOperationExecutor
+        ))))))))))))))))));
+        // @formatter:on
+    }
```

Comment:
Can we somehow keep this Gradle user home-scoped? We don't need the functionality now, but it would be nice to be able toe execute work outside of the scope of builds, maybe some maintenance stuff between builds etc.

Reply:
The thing is I'm unsure why it's not visible right now. If user home scope is "higher" then I should see the execution engine. I don't, which kind of feels like build scope is higher level than user home scope.